main/static/script.js

```diff
@@ -16,20 +19,47 @@ module.controller("mainCtrl", function($scope, $http) {
 		    $scope.queryResult = data;
 		  });
 	};
+
 	$scope.$watch("queryResult", resultUpdate);
+
+	function readHash() {
+		var urlQuery = window.location.hash
+		if (urlQuery != "") {
+			// Drop the leading '#'
+			urlQuery = urlQuery.substring(1);
+			// The remainder is the query to perform.
+			// Decode it (if neccesarry).
+			$scope.queryText = decodeURIComponent(urlQuery);
+			$scope.onSubmitQuery();
+		}
+	}
+
+	readHash(); // Read the current hash (if present).
+
+	// Whenever the user changes the hash (such as pasting in a new URL), call readHash() again.
+	window.addEventListener("hashchange", readHash, false);
```

Comment:
With anchors, it actually doesn't (which is why this is necesarry, but wouldn't be needed if I used `?q=` parameters). For example, going to http://en.wikipedia.org/wiki/Water and clicking on the table of contents doesn't cause the page to reload, even though the URL changed (since only the hash part changed).